client/state/selectors/has-site-comments.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { get, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+
+/**
+ * Returns true if we have site comments
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  siteId      The ID of the site we're querying
+ */
+export const hasSiteComments = createSelector(
+	( state, siteId ) => {
+		const comments = get( state, 'comments.items', {} );
+		return !! find( Object.keys( comments ), key => parseInt( key.split( '-', 1 ), 10 ) === siteId );
+	},
+	state => [ state.comments.items ]
```

Comment:
when I dug into `createSelector` I saw it was doing a shallow-comparison of this but I don't know if we need to do a property-wise comparison. seems like just comparing the reference equality should be enough, no?

that is to say, if we're wanting to memoize this selector, aren't we doing it wrong since Redux should produce a new top-level object when `items` changes? this code will iterate over the new and old items object on every call, which could stack up if we have lots of loaded comments.

not sure if it's a performance issue, but I can't see how adding this dependency is actually helping us